x-pack/plugins/apm/server/lib/services/get_service_infrastructure.ts

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { Setup, SetupTimeRange } from '../helpers/setup_request';
+import { ESFilter } from '../../../../../../src/core/types/elasticsearch';
+import { rangeQuery, kqlQuery } from '../../../../observability/server';
+import { environmentQuery } from '../../../common/utils/environment_query';
+import { ProcessorEvent } from '../../../common/processor_event';
+import {
+  SERVICE_NAME,
+  CONTAINER_ID,
+  POD_NAME,
+} from '../../../common/elasticsearch_fieldnames';
+
+export const getServiceInfrastructure = async ({
+  kuery,
+  serviceName,
+  environment,
+  setup,
+}: {
+  kuery?: string;
+  serviceName: string;
+  environment?: string;
+  setup: Setup & SetupTimeRange;
+}) => {
+  const { apmEventClient, start, end } = setup;
+
+  const filter: ESFilter[] = [
+    { term: { [SERVICE_NAME]: serviceName } },
+    ...rangeQuery(start, end),
+    ...environmentQuery(environment),
+    ...kqlQuery(kuery),
+  ];
+
+  const response = await apmEventClient.search('get_service_infrastructure', {
+    apm: {
+      events: [ProcessorEvent.metric],
+    },
+    body: {
+      size: 0,
+      query: {
+        bool: {
+          filter,
+        },
+      },
+      aggs: {
+        containerIds: {
+          terms: {
+            field: CONTAINER_ID,
+            size: 500,
+          },
+        },
+        hostNames: {
+          terms: {
+            field: 'host.name',
```

Comment:
If `host.name` is different from `host.hostname` can you add it here?

https://github.com/elastic/kibana/blob/b16d20b50aa72dac9a1dbc1f7e3e57b5c24f77ef/x-pack/plugins/apm/common/elasticsearch_fieldnames.ts#L115-L117

Reply:
There seems to have been a lot of back and forth about `host.hostname` and `host.name` over the years, but we do use `host.name` on logs / metrics so I think that makes most sense.

I was going to rename the current `HOST_NAME` to `HOST_HOSTNAME`, but it's in use in quite a few places and one of those is telemetry, so rather than risking anything I've added `HOSTNAME` which doesn't 100% follow the naming style, but seems okay? If it's too confusing I can go with renaming bits.

Reply:
That makes sense. Let's leave it as-is.